### PR TITLE
GLTFExporter: Account for bind matrix.

### LINKED ISSUE #20803: GLTFExporter not accounting for SkinnedMesh bindMatrix
**Describe the bug**

Using `THREE.GLTFExporter` to export a `SkinnedMesh` with a `bindMatrix` that is not the identity matrix results in the `bindMatrix` being lost. If the exported mesh is later imported, 

**To Reproduce**

1. Use `FBXLoader` to load the model from https://sketchfab.com/3d-models/mannequin-anatomy-aid-free-download-2f67c4384e7f4a5b86d84ae9199b7d3b
2. See that mesh is properly aligned with skeleton (See Screenshot 1 below)
3. Export mesh using `THREE.GLTFExporter`
4. Import model exported in step 3
5. See that mesh is improperly aligned with skeleton (See Screenshot 2 below)

***Code***

Fix proposed in: #20804

***Live example***

None at the moment. I can put one together if necessary.

**Expected behavior**

Skeleton should be properly aligned after being exported and re-imported.

**Screenshots**

1. Mesh is properly aligned (on original import):

image

2. Mesh is improperly aligned after importing exported GLB:

image

**Platform:**

 - Device: Any
 - OS: Any
 - Browser: Any
 - Three.js version: Noticed in 0.111.6/A-FRAME 1.0.4. Still appears to be present in dev

### PR DESCRIPTION
**Pull request by zach-capalbo**
Related issue: Fixed #20803

**Description**

Applies `SkinnedMesh` `bindMatrix` to bone inverses when exporting to ensure they retain the correct relation to the mesh.

### CODE REVIEW

**Comment by zach-capalbo** [comment]
Note, the example I've included in #20803 includes a work-around that simulates this fix by multiplying the bone inverses by the skeleton bind matrix before export:

https://glitch.com/edit/#!/transparent-brassy-mercury?path=index.html%3A94%3A37

**Comment by Mugen87** [comment]
Instead of modifiyng skeleton data, couldn't you just honor the bone inverse transform in the glTF node's transformation?

File: examples/js/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -1693,12 +1693,15 @@ THREE.GLTFExporter.prototype = {
...
+			var temporaryBoneInverse = new THREE.Matrix4;
 
 			for ( var i = 0; i < skeleton.bones.length; ++ i ) {
 
 				joints.push( nodeMap.get( skeleton.bones[ i ] ) );
 
-				skeleton.boneInverses[ i ].toArray( inverseBindMatrices, i * 16 );
+				temporaryBoneInverse.copy(skeleton.boneInverses[ i ]);
+				
+				temporaryBoneInverse.multiply(object.bindMatrix).toArray( inverseBindMatrices, i * 16 );
```
Comment (Mugen87) [inline examples/js/exporters/GLTFExporter.js:1704]: Minor style issues:
```js
temporaryBoneInverse.copy( skeleton.boneInverses[ i ] );

temporaryBoneInverse.multiply( object.bindMatrix ).toArray( inverseBindMatrices, i * 16 );
```

File: examples/js/exporters/GLTFExporter.js [hunk 2]
```diff
@@ -1693,12 +1693,15 @@ THREE.GLTFExporter.prototype = {
 
 			var joints = [];
 			var inverseBindMatrices = new Float32Array( skeleton.bones.length * 16 );
+			var temporaryBoneInverse = new THREE.Matrix4;
```
Comment (Mugen87) [inline examples/js/exporters/GLTFExporter.js:1696]: > var temporaryBoneInverse = new THREE.Matrix4;

var temporaryBoneInverse = new THREE.Matrix4();

File: examples/js/exporters/GLTFExporter.js (hunk 1 above)
Comment (zach-capalbo) [reply examples/js/exporters/GLTFExporter.js]: Whoops. Thanks!

**Comment by Mugen87** [comment]
Can you please also run `node utils/modularize.js` once to generate the ES6 module?

**Comment by zach-capalbo** [comment]
> Can you please also run `node utils/modularize.js` once to generate the ES6 module?

Yup. Done now.

File: examples/jsm/exporters/GLTFExporter.js [hunk 3]
```diff
@@ -1716,12 +1716,15 @@ GLTFExporter.prototype = {
 
 			var joints = [];
 			var inverseBindMatrices = new Float32Array( skeleton.bones.length * 16 );
+			var temporaryBoneInverse = new THREE.Matrix4();
```
Comment (Mugen87) [inline examples/jsm/exporters/GLTFExporter.js:1719]: Modules do not use the THREE namespace.

Comment (zach-capalbo) [reply examples/jsm/exporters/GLTFExporter.js:1719]: > Notice this kind of errors does not happen if you use the mentioned script.

`node utils/modularize.js`, correct? I committed the result of running that script. I'll look into why it generated the incorrect output. Perhaps I have something set up incorrectly.

Comment (Mugen87) [reply examples/jsm/exporters/GLTFExporter.js:1719]: > I committed the result of running that script.

Oh dear, `Matrix4` is on the ignore list here:

https://github.com/mrdoob/three.js/blob/6748b35cd034615ef3a45afb6a68f5b66ce301c1/utils/modularize.js#L38

You just have to remove the entry and run the script again.

Sorry for assumption but sometimes users just copy/paste the code from one file into another.

**Review by Mugen87 — APPROVED** [review]

File: examples/jsm/exporters/GLTFExporter.js (hunk 3 above)
Comment (zach-capalbo) [reply examples/jsm/exporters/GLTFExporter.js]: Ok, I've removed `Matrix4` from the ignore list, run the script, and committed the result (including the change to `modularize.js`)

It looks like it also changed an unrelated documentation reference elsewhere in `GLTFExporter`. I'm not sure if this is desirable or not.

**Review by takahirox — APPROVED** [review]
Good catch and this change looks correct to me.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**